src/webserver/DOSGuard.h

```diff
@@ -9,11 +9,47 @@
 class DOSGuard
 {
     boost::asio::io_context& ctx_;
-    std::mutex mtx_;  // protects ipFetchCount_
+    std::recursive_mutex mtx_;  // protects ipFetchCount_, ipRequestCount_
     std::unordered_map<std::string, std::uint32_t> ipFetchCount_;
+    std::unordered_map<std::string, std::uint32_t> ipRequestCount_;
     std::unordered_set<std::string> const whitelist_;
     std::uint32_t const maxFetches_;
     std::uint32_t const sweepInterval_;
+    std::uint32_t const maxConcurrentRequests_;
+
+    struct ticket
```

Comment:
Can we name this something like `RPCScope` or something a little more descriptive.

If you like ticket, lets name this `Ticket`, class names are generally capitalized.